Fix to validate required properties first and not allow rules persistent providers to save entity without required fields

oleksiimiroshnyk opened the pull request:
- Fix to validate required properties first and not allow rules persistent providers to save entity without required fields

Fixes: BroadleafCommerce/QA#4688

Commit:
BroadleafCommerce/QA#4688
- Fix to validate required properties first and not allow rules persistent providers to save entity without required fields

StanislavFedorov reviewed (approved):
For me works fine. Just one note.
I think we can use this
`entity.isValidationFailure()`

instead of this
`entity.getPropertyValidationErrors() != null && entity.getPropertyValidationErrors().isEmpty()`

Commit:
BroadleafCommerce/QA#4688
- Fix to validate required properties first and not allow rules persistent providers to save entity without required fields

danielcolgrove commented:
This fix looks pretty straight forward.  My main concern was how invasive the fix may have been.  If this tests our okay, I'm good with the change.

riteshadhikari17 commented:
@oleksiimiroshnyk I am seeing two error messages. Can you fix this quickly?
![image](https://user-images.githubusercontent.com/35038090/166694202-366e315e-0885-495f-8715-9ed17ed5880f.png)

Commit:
BroadleafCommerce/QA#4688
- Don't validate if we already have some errors, otherwise we can have duplication of the error message

oleksiimiroshnyk commented:
@riteshadhikari17 pushed a fix
